packages/sessions-sdk-react/src/components/notional-amount.tsx

```diff
@@ -0,0 +1,49 @@
+import clsx from "clsx";
+import * as dnum from "dnum";
+
+import { stringToAmount } from "../amount-to-string.js";
+import { calculateNotional } from "../calculate-notional.js";
+import styles from "./notional-amount.module.css";
+
+type Props = {
+  amount: string;
+  decimals: number;
+  price: number;
+  className?: string | undefined;
+  validationError?: string | undefined;
+};
+
+export const NotionalAmount = ({
+  amount,
+  decimals,
+  price,
+  className,
+  validationError,
+}: Props) => {
+  if (validationError) {
+    return (
+      <div className={clsx(className, styles.error)}>
+        Invalid amount specified
+      </div>
+    );
+  }
+
+  if (amount.length === 0) {
```

Comment:
possibly this check belongs in the JSX where `NotionalAmount` is invoked? but in the spirit of deduplication i put this here, even though semantically i think it's better as a check to determine whether `NotionalAmount` is even invoked. open to suggestions here

Reply:
It's a bit weird to pass `amount` as a string for sure but I wouldn't overthink it, this is a reasonable api for how it's used in practice in all current cases.  We can revisit later when we build more features and see how things transform.